### ISSUE TITLE
Missing activity for favorites via WebDAV

### ISSUE THREAD

**Issue by MorrisJobke** [nextcloud/server#26393]
* mark a file as favorite via WebDAV (Android or iOS app for example)
* open the activity view
* expected: "You have favorited FILE"
* actual: no activity

The web UI uses `apps/files/api/v1/files/FILE` API that does trigger the activity.

The code to add the tag:

https://github.com/nextcloud/server/blob/d89a75be0b01f0423a7c1ad2d58aac73c3cc1f3a/apps/files/lib/Service/TagService.php#L85-L112

And then the actitivity:

https://github.com/nextcloud/server/blob/d89a75be0b01f0423a7c1ad2d58aac73c3cc1f3a/apps/files/lib/Service/TagService.php#L119-L148

While the PROPPATCH doesn't trigger anything related to that:

https://github.com/nextcloud/server/blob/eb502c02ff7693bb36318d857985f79e7bac370c/apps/dav/lib/Connector/Sabre/TagsPlugin.php#L284-L295

How to handle this? The ugly way by making tthe method in `TagService` public and call from the dav app directly into the files app to have it all in one place, or to copy it over and deprecated the files app approach as we should go for DAV based API access anyways? I would say option 2.

cc @rullzer @ChristophWurst @kesselb @PVince81 @juliushaertl

**Comment by rullzer** [comment]
It is webdav. So I consider this a core app. I'd be fine to call an internal function.

Long term we should move it all to DAV and kill the dedicatd endpoint IMO

**Comment by MorrisJobke** [comment]
> Long term we should move it all to DAV and kill the dedicatd endpoint IMO

Yep, but then we can also copy it over now and then just deprecate the old method :) Nothing lost and avoided unnecessary complexity that we need to reduce in the future again.

**Comment by rullzer** [comment]
Fine by me :)

**Comment by szaimen** [comment]
Hi, please update to 24.0.9 or better 25.0.3 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 25-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+
